discord/ext/commands/cog.py

```diff
@@ -585,6 +585,16 @@ def cog_check(self, ctx: Context[BotT]) -> bool:
         """
         return True
 
+    @_cog_special_method
+    def interaction_check(self, interaction: discord.Interaction[BotT], /) -> bool:
```

Comment:
This has to be async.

Reply:
But the library uses utils.maybe_coroutine? 
https://github.com/Rapptz/discord.py/blob/f7e97954950ffb0e34238d70813454caa6f1a3ae/discord/app_commands/commands.py#L991-L995